### PR TITLE
Implements `background`

### LINKED ISSUE #247: `background` not implemented
http://www.byond.com/docs/ref/#/proc/set/background

### PR DESCRIPTION
**Pull request by ike709**
It... just works?

Closes #247

### CODE REVIEW

**Comment by wixoaGit** [comment]
Does the correct behavior for background look like this:
```
for(var/i in 1 to 3)
    world.log << i
    sleep(-1)
```
or this?
```
for(var/i in 1 to 3)
    world.log << i
sleep(-1)
```

The DM reference states `It would only ever be interrupted at the end of the for or while loops.`

**Comment by ike709** [comment]
> You could achieve a similar thing by sprinkling calls to sleep(0) or sleep(-1) in the "working" part of the loop.

This implies the former, IMO.

Also it seems like it wouldn't be that useful if it applied _outside_ of the looping

**Comment by ike709** [comment]
I wonder if I should actually do it in `LoopContinue()`

File: DMCompiler/DM/Visitors/DMProcBuilder.cs [hunk 1]
```diff
@@ -57,6 +57,14 @@ class DMProcBuilder {
             foreach (DMASTProcStatement statement in block.Statements) {
                 try
                 {
+                    // ProcessStatementSet() needs to be before any loops
```
Comment (wixoaGit) [inline DMCompiler/DM/Visitors/DMProcBuilder.cs:60]: This doesn't have any affect. It's just moving the `DMASTProcStatementSet` check outside of `ProcessStatement()`, it's still being processed in the same order.

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Review by wixoaGit — APPROVED** [review]
In the future it'd be better to have a sleep opcode instead of depending on the runtime implementing a `sleep()` proc, but this works for now.

**Pull request merged**